src/frontend/tsx/Challenges.tsx

```diff
@@ -77,16 +79,45 @@ export function Challenges () {
     await pokeAction(urbit, challenge(who, side, description), onError, onSuccess)
   }
 
+  const findFriends = async () => {
+    const friendsScry: Object = await urbit.scry({ app: 'chess', path: '/friends' })
+    const stringified: string = JSON.stringify(friendsScry, null, 2)
+    const parsed = JSON.parse(stringified) as { friends: Array<Ship> }
+    setFriends(parsed.friends)
+  }
+
+  const openFriends = () => {
+    setFriendsList(true)
+    findFriends()
+    // XX: set showingIncoming and showingOutgoing to false
+  }
+
+  const quickGame = async (opponent: Ship) => {
+    const onError = () => {
+      incrementBadChallengeAttempts()
+    }
+
+    const onSuccess = () => {
+      resetChallengeInterface()
+    }
+    // XX: fill in a default description,
+    //     maybe current date like '~2022.12.7'
+    //
+    // XX: add this challenge to outgoingChallenges,
+    //     handle UI to show we can't challenge someone
+    //     who we've already challenged
+    await pokeAction(urbit, challenge('~' + opponent, Side.Random, ''), onError, onSuccess)
+  }
```

Comment:
Would it be difficult to instead have the quick challenge button open the same pop-up that "New Challenge" opens, but pre-populate it with the friendly ship's `@p`? Would solve the problems of only being able to challenge as random, needing a default description, and replicating success/failure logic.